pkg/cluster/cluster.go

```diff
@@ -255,16 +256,28 @@ func setOptionsDefaults(options Options) Options {
 // NewClientFunc allows a user to define how to create a client.
 type NewClientFunc func(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error)
 
-// DefaultNewClient creates the default caching client.
+// DefaultNewClient creates the default caching client, that will not cache Unstructured.
 func DefaultNewClient(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
+	return newClient(cache, config, options, false, uncachedObjects)
+}
+
+// CacheUnstructuredNewClient creates a caching client that will cache Unstructured.
+func CacheUnstructuredNewClient(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
```

Comment:
How about we do this similiar to what we do in the cache, where we have a `func BuilderWithOptions(options Options) NewCacheFunc {` to accomodate the case that ppl might want additional options in the future?

Reply:
For example, we have for a long time discussed that the default behavior of the cache-backed client of automatically creating caches is not great, but it is extremely hard to change due to compatibility converns (ppl might only discover this change at runtime).

So having an option to allow ppl to opt into this would be nice

Reply:
@alvaroaleman Thanks a lot for the initial review and your suggestion. I have made an attempt to implement the suggested design with an extensible `cluster.ClientOptions` struct. Please take another look, and tell me what you think.